src/aap_eda/core/utils/credentials.py

```diff
@@ -446,3 +455,18 @@ def _validate_file_template_key(key: str, key_names: list[str]) -> None:
                     )
                     % {"injector_type": "file", "key": key}
                 )
+
+
+def check_reserved_keys_in_extra_vars(data: dict[str, any]) -> None:
+    for key in data.keys():
+        if key in RESERVED_EXTRA_VAR_KEYS:
+            raise ValidationError(
+                _(
+                    "Extra vars key %(key)s cannot be one of these "
+                    "reserved keys %(reserved)s"
```

Comment:
I find it preferable to visually offset both offending key and reserved keys to make them stand out as the salient portions of the message.  E.g.,
`"Extra vars key '%(key)s' cannot be one of these "
                     "reserved keys '%(reserved)s'"`